### ISSUE TITLE
nostd?

### ISSUE THREAD

**Issue by InfiniteCoder01** [gregorygaines/bitfields-rs#4]
Can this crate work in nostd environment? If so, std feature should be added. It would be very useful for OS development

**Comment by InfiniteCoder01** [comment]
For any wondering, I found [this crate](https://crates.io/crates/bitfield-struct), which should be usable in nostd

**Comment by gregorygaines** [comment]
Yes, the bitfields crate works in no-std enviornments. In fact, here's a test for it: https://github.com/gregorygaines/bitfields-rs/blob/main/bitfields/tests/compile_error_cases/bitfield_no_std.rs.

Yes, bitfield-struct is a wonderful crate that is usable in no-std.

**Comment by gregorygaines** [comment]
Following back up on this, there was a user that wasn't able to use this crate in no-std. I added a fix that hopefully fixes no_std issues for users: https://github.com/gregorygaines/bitfields-rs/issues/8.

Thanks.
